internal/state/rollback.go

```diff
@@ -0,0 +1,86 @@
+package state
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/tendermint/tendermint/version"
+)
+
+// Rollback takes a state at height n and overwrites it with the state
+// at height n - 1. Note state here refers to tendermint state not application state.
+func Rollback(bs BlockStore, ss Store) (int64, []byte, error) {
+	invalidState, err := ss.Load()
+	if err != nil {
+		return -1, []byte{}, err
```

Comment:
These `[]byte{}` can all be `nil`.

```suggestion
		return -1, nil err
```

```sh
gofmt -w -r '[]byte{}->nil' internal/state/rollback.go
```